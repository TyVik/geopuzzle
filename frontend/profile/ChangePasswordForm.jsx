'use strict';
import React from "react";
import {Alert, Button, Form} from "react-bootstrap";
import {CSRFfetch, getFormData} from "../utils";
import {Input} from '../components/Input';
import {Field, Form as FormWrapper} from "react-final-form";
import {FormattedMessage as Msg} from "react-intl";


export default class ChangePasswordForm extends React.Component {
  onSubmit = async (values, form, callback) => {
    let data = getFormData(values);
    data.append('new_password2', values['new_password1']);
    let options = {method: 'POST', body: data};
    let response = await CSRFfetch(`${window.location.pathname}?section=password`, options);
    let result = await response.json();
    if (Object.keys(result).length === 0) {
      form.reset();
    }
    return result;
  };

  _render = ({handleSubmit, form, submitting}) => {
    let state = form.getState();
    return <Form onSubmit={handleSubmit}>
      {state.submitSucceeded && <Alert variant="success"><Msg id="password.changed"/></Alert>}
      <Field name="old_password" component={Input} label={<Msg id="password.current" type="password"/>}/>
      <Field name="new_password1" component={Input} label={<Msg id="password.new"/>} type="password"/>
      <Button variant="primary" type="submit" ><Msg id="password.change"/></Button>
      {submitting && <div className="spinner-border" role="status" />}
    </Form>;
  };

  render() {
    return <FormWrapper onSubmit={this.onSubmit} render={this._render} />;
  }
}
