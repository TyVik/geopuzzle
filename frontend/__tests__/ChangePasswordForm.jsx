import React from 'react';
import {shallow} from 'enzyme';
import {Button} from "react-bootstrap";
import ChangePasswordForm from '../profile/ChangePasswordForm';


describe('shallow <ChangePasswordForm /> components', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ChangePasswordForm/>);
  });

  it('render empty', () => {
    expect(wrapper).toMatchSnapshot('ChangePasswordForm');
  });

  it('render submit fail', () => {
    wrapper.find(Button).simulate('click');
    expect(wrapper).toMatchSnapshot('ChangePasswordForm');
  });

/*  it('check props', () => {
    expect(wrapper).toMatchSnapshot('infobox');
    expect(shallow(<Infobox show={true} coat_of_arms='coat_of_arms'/>)).toMatchSnapshot('infobox-empty');
  });

  it('click close', () => {
    wrapper.find('button.close').simulate('click');
    expect(onClose).toHaveBeenCalled();
  });

  it('click collapse', () => {
    wrapper.find('.header i').simulate('click');
    expect(localStorage.getItem('infobox_collapse')).toBe('true');
    expect(wrapper).toMatchSnapshot('infobox-collapse');
    wrapper.find('.header i').simulate('click');
    expect(localStorage.getItem('infobox_collapse')).toBe('false');
    expect(wrapper).toMatchSnapshot('infobox');
  });*/
});
