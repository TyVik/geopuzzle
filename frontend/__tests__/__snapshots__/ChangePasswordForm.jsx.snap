// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow <ChangePasswordForm /> components check props: infobox 1`] = `undefined`;

exports[`shallow <ChangePasswordForm /> components render empty: ChangePasswordForm 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChangePasswordForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "onSubmit": [Function],
      "render": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "onSubmit": [Function],
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
